packages/vm/test/t8n/helpers.ts

```diff
@@ -124,3 +133,71 @@ export function normalizeNumbers(input: any) {
   }
   return input
 }
+
+/**
+ * Formats an individual EVM step trace as a JSON object
+ * @param step an {@link InterpreterStep} emitted by the EVM `step` event
+ * @param memory whether to include the memory in the trace
+ * @returns a JSON object that matches the EIP-7756 trace format
+ */
+export const stepTraceJSON = (
+  step: InterpreterStep,
+  memory: boolean = false,
+  storage: boolean = false,
+) => {
+  let hexStack = []
+  hexStack = step.stack.map((item: bigint) => {
+    return '0x' + item.toString(16)
+  })
+  let memWords = undefined
+  const memSize = Number(step.memoryWordCount) * 8 // memSize is reported in bytes, not words (i.e 32 bytes)
+  if (memory) {
+    memWords = []
+    for (let i = 0; i < step.memoryWordCount; i++) {
+      memWords.push(bytesToHex(step.memory.slice(i * 32, (i + 1) * 32))) // memory is returned in 32 byte words
+    }
+  }
+
+  let storageWords = undefined
+  if (storage) {
+    storageWords = []
+    for (let i = 0; i < step.memoryWordCount; i++) {
+      storageWords.push(bytesToHex(step.memory.slice(i * 32, (i + 1) * 32))) // memory is returned in 32 byte words
+    }
+  }
+  const opTrace = {
+    pc: step.pc,
+    op: '0x' + step.opcode.code.toString(16),
+    gas: Number(step.gasLeft),
+    gasCost: Number(step.opcode.dynamicFee ?? BigInt(step.opcode.fee)), // if `dynamicFee` is set, it includes base fee
+    memory: memory ? memWords : undefined,
+    memSize,
+    stack: hexStack,
+    depth: step.depth + 1, // Depth starts at 1 - EIP-7756
+    refund: Number(step.gasRefund),
+    opName: step.opcode.name,
+    section: step.eofSection,
+    immediate: step.immediate !== undefined ? bytesToHex(step.immediate) : undefined,
+    functionDepth: step.eofFunctionDepth,
+    error: step.error !== undefined ? step.error.toString() : undefined,
+    storage: storage ? step.storage : undefined,
+  }
+  return opTrace
+}
+
+/**
+ * Formats an individual EVM summary trace as a JSON object
+ * @param event an {@link AfterTxEvent} emitted by the vm `afterTx` event
+ * @param vm a {@link VM} instance
+ * @returns a JSON object that matches the EIP-7756 summary object format
+ */
+export const summaryTraceJSON = async (event: AfterTxEvent, vm: VM) => {
```

Comment:
Should we add this to the output of `afterTx`? I think we should!

Reply:
I'm not really sure I understand.  The `afterTx` event isn't directly translatable to the Summary event

Reply:
Ah no, I meant adding a summary item to the object which gets reported on the `afterTx`. This is fine for now though.